packages/nimble-components/src/table-column/icon/models/table-column-icon-validator.ts

```diff
@@ -23,22 +24,43 @@ export class TableColumnIconValidator extends TableColumnEnumBaseValidator<
         super(columnInternals, iconValidityFlagNames, [
             MappingIcon,
             MappingSpinner
-        ]);
+        ] as const);
+    }
+
+    private static isIconMappingElement(
+        mapping: Mapping<unknown>
+    ): mapping is MappingIcon {
+        return mapping instanceof MappingIcon;
+    }
+
+    private static isSupportedMappingElement(
```

Comment:
Can we make this non-`static` so it can use `supportedMappingElements` from the base? We would also have to change that from `private` to `protected`. Should have single source of truth.

Reply:
Like the sentiment of single sourcing the knowledge of supported elements. I tried to make the typing stronger for giving the base class a list of inferred supported Mappings but that was tricky. I instead went the other way and just made each specific implementation responsible for managing their supported mappings. Think it looks pretty good and isn't a concerning amount of duplication. I'll resolve but reset you in-case there are concerns.